src/app/subjects/subject-table/subject-table.component.ts

```diff
@@ -0,0 +1,105 @@
+import { Component, OnInit } from '@angular/core';
+import { FormControl } from '@angular/forms';
+import { ActivatedRoute, ParamMap } from '@angular/router';
+import { MatDatepickerInputEvent } from '@angular/material/datepicker';
+
+import { Subscription } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
+
+import { BaseComponent } from 'src/app/components/base/base.component';
+import { ITableConfig, ICell, IChangeField } from 'src/app/common/models/Table';
+import { Subject } from '../../common/models/Subject';
+
+import { SubjectService } from '../services/subject.service';
+import { SubjectTableService } from '../services/subject-table.service';
+import { SubjectTableConfigService } from '../services/subject-table-config.service';
+import { SubjectTableHeaderService } from '../services/subject-table-header.service';
+import { SubjectTableBodyService } from '../services/subject-table-body.service';
+
+@Component({
+  selector: 'app-subject-table',
+  templateUrl: './subject-table.component.html',
+  styleUrls: ['./subject-table.component.scss'],
+  providers: [
+    SubjectTableService,
+    SubjectTableConfigService,
+    SubjectTableHeaderService,
+    SubjectTableBodyService,
+  ],
+})
+export class SubjectTableComponent extends BaseComponent implements OnInit {
+  public subject: Subject;
+  public config: ITableConfig<ICell<string>>;
+  public teacherControl: FormControl;
+
+  constructor(
+    private route: ActivatedRoute,
+    private tableService: SubjectTableService,
+    private subjectService: SubjectService,
+  ) {
+    super();
+  }
+
+  private getSubject(name: string): void {
+    this.tableService
+      .fetchSubject(name)
+      .pipe(takeUntil(this.unsubscribe$))
+      .subscribe({
+        next: (subject: Subject) => {
+          this.subject = subject || new Subject();
```

Comment:
What for is `|| new Subject()`. Even if it is needed move it to ` next: (subject: Subject = new Subject()) => {`